tests/ops/qubit/test_matrix_ops.py

```diff
@@ -269,16 +279,30 @@ def test_qubit_unitary_decomposition(self, U, expected_gate, expected_params):
         assert isinstance(decomp2[0], expected_gate)
         assert np.allclose(decomp2[0].parameters, expected_params, atol=1e-7)
 
-    def test_error_qubit_unitary_decomposition_broadcasted(self):
+    def test_qubit_unitary_decomposition_calls_zyz_decomposition(self):
+        """Tests that single-qubit QubitUnitary decompositions call zyz_decomposition."""
+
+        with patch.object(
+            qml.transforms.decompositions, "zyz_decomposition"
+        ) as mocked_zyz_decomposition:
+            decomp = qml.QubitUnitary.compute_decomposition(H, wires=0)
+            np.testing.assert_equal(mocked_zyz_decomposition.call_args.args[0], H)
+            assert mocked_zyz_decomposition.call_args.args[1] == 0
+
+            decomp2 = qml.QubitUnitary(X, wires=1).decomposition()
+            np.testing.assert_equal(mocked_zyz_decomposition.call_args.args[0], X)
+            assert mocked_zyz_decomposition.call_args.args[1] == 1
+
+            assert mocked_zyz_decomposition.call_count == 2
```

Comment:
I don't think we need to test whether or not `zyz_decomposition` was actually called.

What we need to check is whether it gives the right answer, not necessarily "how" it got to that right answer.

While they can be helpful, I often find mocking and patching causes more trouble then they're worth outside of specific situations where we have no other options.

If we know `decomposition` gives the right answer, we are properly tested.

```suggestion
```